virt-v2v/cold/customize-image.go

```diff
@@ -0,0 +1,121 @@
+package main
+
+import (
+	"embed"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+)
+
+//go:embed scripts
+var scriptFS embed.FS
+
+const (
+	WIN_FIRSTBOOT_PATH = "/Program Files/Guestfs/Firstboot/"
+)
+
+// CustomizeWindows customizes a windows disk image by uploading scripts and configuring new RunOnce task to execute it.
+//
+// The function writes two bash scripts to the specified local tmp directory,
+// uploads them to the disk image using `virt-customize`.
+//
+// NOTE: We can't use the firstboot commands as the system is not ready to run all commands.
+// Some commands for example `Get-Disk` returns empty list. So we need to set the registry with `RunOnce` task,
+// which will be run by the Winodws once the system is ready.
+//
+// Arguments:
+//   - domain (string): The VM which should be customized.
+//
+// Returns:
+//   - error: An error if something goes wrong during the process, or nil if successful.
+func CustomizeWindows(domain string) error {
+	fmt.Printf("Customizing domain '%s'", domain)
+	t := EmbedTool{filesystem: &scriptFS}
+	err := t.CreateFilesFromFS(DIR)
+	if err != nil {
+		return err
+	}
+	windowsScriptsPath := filepath.Join(DIR, "scripts", "windows")
+	initPath := filepath.Join(windowsScriptsPath, "restore_config_init.bat")
+	restoreScriptPath := filepath.Join(windowsScriptsPath, "restore_config.ps1")
+
+	// Upload scripts to the windows
+	uploadScriptPath := fmt.Sprintf("%s:%s", restoreScriptPath, WIN_FIRSTBOOT_PATH)
+	uploadInitPath := fmt.Sprintf("%s:%s", initPath, WIN_FIRSTBOOT_PATH)
+
+	var extraArgs []string
+	extraArgs = append(extraArgs, getScriptArgs("upload", uploadScriptPath, uploadInitPath)...)
```

Comment:
```suggestion
	// Get the list of .ps1 and .bat files in the directory
	files, err := os.ReadDir(windowsScriptsPath)
	if err != nil {
		fmt.Printf("Failed to read directory: %v\n", err)
		return
	}

	var extraArgs []string

	for _, file := range files {
		if !file.IsDir() && (strings.HasSuffix(file.Name(), ".ps1") || strings.HasSuffix(file.Name(), ".bat")) {
			// Construct the local file path and the upload path
			localFilePath := filepath.Join(windowsScriptsPath, file.Name())
			uploadPath := fmt.Sprintf("%s:%s", localFilePath, WIN_FIRSTBOOT_PATH)

			// Append the arguments
			extraArgs = append(extraArgs, getScriptArgs("upload", uploadPath)...)
		}
	}
```

This will upload all .ps1 files, no need to update when adding more scripts ( for example mounting script config map at the scripts path ) WDYT ?

Reply:
I \was thinking about this but I wanted to keep it modular as we might want some scripts in different paths or overwrite some files. But definitely could visit this in future